File: BSM208 - Programlama Dillerinin Prensipleri/Uygulamalar/Java/Ders10/src/cc/Kare.java
package cc;

public class Kare extends GeometrikSekil {
	private int kenar;
	
	public Kare(String renk, int kenar) {
		super(renk);
		this.kenar=kenar;
	}

	@Override
	public double Alan() {
		return Math.pow(kenar, 2);
	}

	@Override
	public double Cevre() {
		return 4*kenar;
	}
}

File: BSM208 - Programlama Dillerinin Prensipleri/Uygulamalar/C/Hafta10/src/Kare.c
#include "Kare.h"

Kare KareOlustur(char* renk, int kenar){
	Kare this;
	this=(Kare)malloc(sizeof(struct KARE));
	this->super=GeometrikSekilOlustur(renk);
	this->kenar=kenar;
	
	this->super->Alan=&alan;
	this->super->Cevre=&cevre;
	this->yokEt=&KareYoket;
	return this;
}
double alan(const Kare this){
	return pow(this->kenar, 2);
}
double cevre(const Kare this){
	return 4*this->kenar;
}
void KareYoket(Kare this){
	if(this==NULL)	return;
	this->super->yokEt(this->super);//burada fonksiyon göstericisi yardımıyla geometriksekil'in yıkıcısını çağırmak önemli!!!!
	free(this);
}